extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Services} from '../../../src/services';
+import {SocialShare} from './social-share';
+import {addParamsToUrl} from '../../../src/url';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {getSocialConfig} from './amp-social-share-config';
+import {isExperimentOn} from '../../../src/experiments';
+import {toggle} from '../../../src/style';
+
+/** @const {string} */
+const TAG = 'amp-social-share';
+
+class AmpSocialShare extends PreactBaseElement {
+  /** @override */
+  init() {
+    const viewer = Services.viewerForDoc(this.element);
+    const platform = Services.platformFor(window);
+    const typeConfig = this.getTypeConfigOrUndefined_(
+      this.element.getAttribute('type'),
+      viewer,
+      platform
+    );
+    // Hide/ignore component if typeConfig is undefined
+    if (!typeConfig) {
+      toggle(this.element, false);
+      return;
```

Comment:
Throw here, to make Resources mark this element as errored.

Reply:
Added a warning aligned with v1: "Skipping obsolete share button"